src/pages/Swap/Swap.tsx

```diff
@@ -1,25 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
+import BigNumber from 'bignumber.js';
 import './Swap.css';
 import Mobile from '../NoMobileSupport/Mobile';
 import { handleInstall } from '../../utils';
+import ErrorMessage from '../../components/ErrorMessage';
+import useTdexFetcher from '../../utils/tdexFetcherHooks';
+import { AmountPreview } from '../../utils/rates';
 
 interface SwapProp {
   selectToken: any;
-  checkedCoin: any;
-  checkCoinBottom: any;
+  sendCoin: any;
+  receiveCoin: any;
   isInstalled: boolean;
   isConnected: boolean;
+  changeToken: any;
+  selectError: string | null;
   isMobile: boolean;
 }
 
 export const Swap: React.FC<SwapProp> = ({
   selectToken,
-  checkedCoin,
-  checkCoinBottom,
+  sendCoin,
+  receiveCoin,
   isInstalled,
   isConnected,
+  changeToken,
+  selectError,
   isMobile,
 }) => {
+  const [changeFlag, setChangeFlag] = useState<boolean>(false);
+  const [amountToBeSent, setAmountToBeSent] = useState<string>('');
+  const [amountToReceive, setAmountToReceive] = useState<string>('');
+  const [isPreviewing, setIsPreviewing] = useState<boolean>(false);
+  const [previewValueError, setPreviewValueError] = useState<Error | null>(
+    null,
+  );
+  const tdexFetcher = useTdexFetcher();
```

Comment:
Checks for `undefined`. It means there is an error to instance the fetcher therefore we should show an error and hide the entire swap form.

Reply:
fixed this